packages/core/src/routes-me/user.ts

```diff
@@ -129,16 +130,27 @@ export default function userRoutes<T extends AuthedMeRouter>(
 
   router.post(
     '/password',
-    koaGuard({ body: object({ password: string().min(1) }) }),
+    koaGuard({ body: object({ password: string().min(1) }), status: [204, 400] }),
     async (ctx, next) => {
       const { id: userId } = ctx.auth;
       const { password } = ctx.guard.body;
 
-      const { isSuspended } = await findUserById(userId);
+      const user = await findUserById(userId);
 
-      assertThat(!isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
+      assertThat(!user.isSuspended, new RequestError({ code: 'user.suspended', status: 401 }));
 
-      await checkVerificationStatus(userId, null);
+      const [signInExperience] = await Promise.all([
+        findDefaultSignInExperience(),
+        checkVerificationStatus(userId, null),
+      ]);
+      const passwordPolicyChecker = new PasswordPolicyChecker(signInExperience.passwordPolicy);
+      const issues = await checkPasswordPolicyForUser(passwordPolicyChecker, password, user);
+
+      if (issues.length > 0) {
+        ctx.status = 400;
+        ctx.body = { issues };
+        return next();
+      }
```

Comment:
Why don't we throw RequestError? The frontend needs to handle the error and display error messages